cloudinit/distros/debian.py

```diff
@@ -155,7 +164,72 @@ def _get_localhost_ip(self):
     def set_timezone(self, tz):
         distros.set_etc_timezone(tz=tz, tz_file=self._find_tz_file(tz))
 
+    def _is_apt_lock_available(self, lock_files=None):
+        """Determines if another process holds any apt locks.
+
+        If all locks are clear, return True else False.
+        """
+        if lock_files is None:
+            lock_files = APT_LOCK_FILES
+        with ExitStack() as stack:
+            file_handles = []
+            for lock in lock_files:
+                try:
+                    file_handles.append(stack.enter_context(open(lock, 'w')))
+                except FileNotFoundError:
+                    # If the lock file doesn't even exist yet, then we don't
+                    # have to wait for it
+                    pass
+            for handle in file_handles:
+                try:
+                    fcntl.lockf(handle, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                except IOError:
+                    return False
+        return True
+
+    def _wait_for_apt_install(
```

Comment:
rename? since this is broader than install?
```suggestion
    def _wait_for_apt_lock(
```